src/DotVVM.Framework/Hosting/DotvvmPresenter.cs

```diff
@@ -392,23 +393,25 @@ public async Task ProcessStaticCommandRequest(IDotvvmRequestContext context)
                 await filter.OnCommandExecutingAsync(context, action);
             }
 
-            object? result = null;
             try
             {
-                Task? resultTask = null;
+                var commandResultOrNotYetComputedAwaitable = action.Action();
 
-                result = action.Action();
-
-                resultTask = result as Task;
-                if (resultTask != null)
+                if (commandResultOrNotYetComputedAwaitable is Task commandTask)
                 {
-                    await resultTask;
+                    await commandTask;
+                    return TaskUtils.GetResult(commandTask);
                 }
 
-                if (resultTask != null)
+                var resultType = commandResultOrNotYetComputedAwaitable?.GetType();
+                var possibleResultAwaiter = resultType?.GetMethod(nameof(Task.GetAwaiter));
```

Comment:
```suggestion
                var possibleResultAwaiter = resultType?.GetMethod(nameof(Task.GetAwaiter), Array.Empty<Task>());
```

The awaiter method must have no parameters - I'd add that to the check to prevent false positives of the check,